metrics/time/launch_times.sh

```diff
@@ -68,105 +82,124 @@ sn_to_ns() {
 
 # convert 'nanoseconds' (since epoch) into a 'float' seconds
 ns_to_s() {
-	printf "%.03f" $(bc <<< "scale=3; $1 / 1000000000")
+	printf "%.0${CALC_SCALE}f" $(bc <<< "scale=$CALC_SCALE; $1 / 1000000000")
 }
 
 run_workload() {
-	start_time=$($DATECMD)
-
-	# Check entropy level of the host
-	check_entropy_level
-
-	# Run the image and command and capture the results into an array...
-	declare workload_result
-	readarray -n 0 workload_result < <(sudo -E "${CTR_EXE}" run --rm --runtime=${CTR_RUNTIME} ${IMAGE} test bash -c "$DATECMD $DMESGCMD")
-
-	end_time=$($DATECMD)
-
-	# Delay this calculation until after we have run - do not want
-	# to measure it in the results
-	start_time=$(sn_to_ns $start_time)
-	end_time=$(sn_to_ns $end_time)
-
-	# Extract the 'date' info from the first line of the log
-	# This script assumes the VM clock is in sync with the host clock...
-	workload_time=${workload_result[0]}
-	workload_time=$(echo $workload_time | tr -d '\r')
-	workload_time=$(sn_to_ns $workload_time)
-
-	# How long did the whole launch/quit take
-	total_period=$((end_time-start_time))
-	# How long did it take to get to the workload
-	workload_period=$((workload_time-start_time))
-	# How long did it take to quit
-	shutdown_period=$((end_time-workload_time))
-
-	if [ -n "$CALCULATE_KERNEL" ]; then
-		# Grab the last kernel dmesg time
-		# In our case, we need to find the last real kernel line before
-		# the systemd lines begin. The last:
-		# 'Freeing unused kernel' line is a reasonable
-		# 'last in kernel line' to look for.
-		# We make a presumption here that as we are in a cold-boot VM
-		# kernel, the first dmesg is at '0 seconds', so the timestamp
-		# of that last line is the length of time in the kernel.
-		kernel_last_line=$( (fgrep "Freeing unused kernel" <<- EOF
-				${workload_result[@]}
-			EOF
-			) | tail -1 )
-
-		if [ -z "$kernel_last_line" ]; then
-			echo "No kernel last line"
-			for l in "${workload_result[@]}"; do
-				echo ">: [$l]"
-			done
-			die "No kernel last line"
-		fi
-
-
-		kernel_period=$(echo $kernel_last_line | awk '{print $2}' | tr -d "]")
-
-		# And we can then work out how much time it took to get to the kernel
-		to_kernel_period=$(printf "%0f" $(bc <<<"scale=3; $(ns_to_s $workload_period) - $kernel_period"))
-	else
-		kernel_period="0.0"
-		to_kernel_period="0.0"
-	fi
+    start_time=$($DATECMD)
```

Comment:
@dborquez seems like you need to fix the indentation

Reply:
@thank you @GabyCT, now fixed